File: online-admission-system-frontend/online-admission-system-ui/src/Services/PaymentService.js
import axios from 'axios';
 
const API_BASE_URL = 'http://localhost:8080/payments';
 
export const postPayment = (applicationId, payment) => {
    return axios.post(`${API_BASE_URL}/applicationId/${applicationId}`, payment);
};
 
export const getAllPayment = () => {
    return axios.get(API_BASE_URL);
};
 
export const getPaymentDetailsByPaymentId = (paymentId) => {
    return axios.get(`${API_BASE_URL}/${paymentId}`);
};
 
export const getPaymentDetailsByApplicationId = (applicationId) => {
    return axios.get(`${API_BASE_URL}/application/${applicationId}`);
};
 
export const getPaymentDetailsByEmailId = (emailId) => {
    return axios.get(`${API_BASE_URL}/email/${emailId}`);
};
 
export const getPaymentDetailsByStatus = (paymentStatus) => {
    return axios.get(`${API_BASE_URL}/status/${paymentStatus}`);
};
 
export const deletePaymentById = (paymentId) => {
    return axios.delete(`${API_BASE_URL}/${paymentId}`);
};
 
export const updatePaymentDetails = (payment, paymentId) => {
    return axios.put(`${API_BASE_URL}/${paymentId}`, payment);
};

File: online-admission-backend/src/main/java/com/cg/admission/service/PaymentService.java
package com.cg.admission.service;

import java.util.List;

import com.cg.admission.dto.PaymentDto;
import com.cg.admission.entity.Payment;

public interface PaymentService {
	PaymentDto addPayment(Long paymentId, Payment payment);

	List<PaymentDto> viewAllPaymentDetails();

	PaymentDto getPaymentDetailsByPaymentId(Long paymentId);

	PaymentDto getPaymentDetailsByApplicationId(Long applicationId);

	PaymentDto getPaymentDetailsByEmailId(String emailId);

	String deletePaymentById(Long paymentId);

	PaymentDto updatePaymentDetails(Payment payment, Long paymentId);

	List<PaymentDto> getPaymentDetailsByStatus(String paymentStatus);

}